### ISSUE TITLE
Allow usage of existing secrets

### ISSUE THREAD

**Issue by fastlorenzo** [Mailu/helm-charts#132]
**Is your feature request related to a problem? Please describe.**
The configuration of secret values in the helm chart is only possible by using variables in the values.yaml file.

**Describe the solution you'd like**
It would be great to have a possibility to reference existing `Secret` in the different part of the helm chart config.

Example:

```yaml
# The secret key is required for protecting authentication cookies and must be set individually for each deployment
# secretKey: chang3m3!

# Use an existing secret for the secret key.
key:
  existingSecret: ""
  secretKey: secret-key
```

Where `secret-key` would look like:

```yaml
apiVersion: v1
data:
  secret-key: <<SECRET>>
kind: Secret
metadata:
  name: secret-key
  namespace: mail
type: Opaque
```

I've made some changes in a fork (https://github.com/fastlorenzo/helm-charts-1/tree/master/mailu) that allows to make use of existing secret for the `secretKey` variable. The only issue I have at the moment is I had to drop support for helm2, which doesn't have the `lookup` functionality (to allow checking if the secret exists in the cluster at install time).
I've made those changes based on the example of the postgres chart from Bitnami (https://github.com/bitnami/charts/tree/master/bitnami/postgresql).

Variables that should be allowed to be set via existing secret (IMO):

- secretKey
- initialAccount.*
- mysql.*
- postgresql.*

**Describe alternatives you've considered**
I'm using flux to maintain my k8s infrastructure in sync with a git repo, which is based on Kustomization. At the moment, I'm using variables stored in secrets for the entire cluster that are replaced on deploy time in the values.yaml via Kustomization. This is a good workaround but having the possibility to use existing secrets would be a better option.

**Additional context**
N/A

**Comment by micw** [comment]
Hello,
thank you for contributing.
I have not seen the "lookup" in your repo, where can I see that code?

**Comment by fastlorenzo** [comment]
great, I'll check if I have time to make a PR for this then 😄 

The lookup is here: https://github.com/fastlorenzo/helm-charts-1/blob/227769ce224fbc5d30f6d1be14987f73b32f3b5a/mailu/templates/_helpers.tpl#L75

**Comment by micw** [comment]
I see. It's great that there is finally a solution for that issue which persisted ~5 years (https://github.com/helm/helm/issues/3053).

**Comment by micw** [comment]
@fastlorenzo Just released chart 0.3.0 for mailu 1.9.x. It "officially" drops helm2 support. Would you mind to create a PR for your request?

**Comment by fastlorenzo** [comment]
Nice, glad to see that! Give me couple of days to work on it and I'll make a PR to handle sensitive values in secrets

**Comment by fastlorenzo** [comment]
Fixed with #193
